src/stringProcessing/relevantWords.js

```diff
@@ -1,312 +1,188 @@
-import { forEach } from "lodash-es";
-import { has } from "lodash-es";
-import { flatMap } from "lodash-es";
-import { values } from "lodash-es";
-import { take } from "lodash-es";
-import { includes } from "lodash-es";
-import { intersection } from "lodash-es";
-import { isEmpty } from "lodash-es";
+import { get } from "lodash-es";
 import { uniq } from "lodash-es";
 
-const densityLowerLimit = 0;
-const densityUpperLimit = 0.03;
-const relevantWordLimit = 100;
-const wordCountLowerLimit = 200;
-
-// First four characters: en dash, em dash, hyphen-minus, and copyright sign.
-const specialCharacters = [ "–", "—", "-", "\u00a9", "#", "%", "/", "\\", "$", "€", "£", "*", "•", "|", "→", "←", "}", "{", "//", "||", "\u200b" ];
-
-/**
- * Returns the word combinations for the given text based on the combination size.
- *
- * @param {string} text The text to retrieve combinations for.
- * @param {number} combinationSize The size of the combinations to retrieve.
- * @param {Function} functionWords The function containing the lists of function words.
- * @returns {WordCombination[]} All word combinations for the given text.
- */
-function getWordCombinations( text, combinationSize, functionWords ) {
-	const sentences = getSentences( text );
-
-	let words, combination;
-
-	return flatMap( sentences, function( sentence ) {
-		sentence = sentence.toLocaleLowerCase();
-		sentence = normalizeQuotes( sentence );
-		words = getWords( sentence );
-
-		return filter( map( words, function( word, i ) {
-			// If there are still enough words in the sentence to slice of.
-			if ( i + combinationSize - 1 < words.length ) {
-				combination = words.slice( i, i + combinationSize );
-				return new WordCombination( combination, 0, functionWords );
-			}
-
-			return false;
-		} ) );
-	} );
-}
-
-/**
- * Calculates occurrences for a list of word combinations.
- *
- * @param {WordCombination[]} wordCombinations The word combinations to calculate occurrences for.
- * @returns {WordCombination[]} Word combinations with their respective occurrences.
- */
-function calculateOccurrences( wordCombinations ) {
-	const occurrences = {};
-
-	forEach( wordCombinations, function( wordCombination ) {
-		const combination = wordCombination.getCombination();
+import getWords from "../stringProcessing/getWords";
+import { normalizeSingle } from "../stringProcessing/quotes";
+import WordCombination from "../values/WordCombination";
+import functionWordListsFactory from "../helpers/getFunctionWords";
+import getStemForLanguageFactory from "../helpers/getStemForLanguage";
 
-		if ( ! has( occurrences, combination ) ) {
-			occurrences[ combination ] = wordCombination;
-		}
-
-		occurrences[ combination ].incrementOccurrences();
-	} );
-
-	return values( occurrences );
-}
+const functionWordLists = functionWordListsFactory();
+const stemFunctions = getStemForLanguageFactory();
+const digitsAndPunctuation = /[1234567890‘’“”"'.…?!:;,¿¡«»&*@#±^%$|~=+§`[\](){}⟨⟩<>/\\–\-\u2014\u00d7\s]/g;
```

Comment:
I like the previous name more. ```¡«»&*@#±^%$|~=+§`[\](){}⟨⟩<>``` are not punctuation, right?

Reply:
Changed it back to `specialCharacters` 😄